filewatcher/filewatcher.go

```diff
@@ -379,3 +393,12 @@ func (fw *MockFileWatcher) Get() interface{} {
 func (fw *MockFileWatcher) Stop() {}
 
 var _ FileWatcher = (*MockFileWatcher)(nil)
+
+func isDirectory(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	return fileInfo.IsDir(), err
```

Comment:
```suggestion
	return fileInfo.IsDir(), nil
```
Return nil error explicitly here.